index.js

```diff
@@ -11,7 +11,51 @@ function toObject(val) {
 	return Object(val);
 }
 
-module.exports = Object.assign || function (target, source) {
+function shouldUseNative() {
+	try {
+		if (!Object.assign) {
+			return false;
+		}
+
+		// Detect buggy property enumeration order in older V8 versions.
+
+		// https://bugs.chromium.org/p/v8/issues/detail?id=4118
+		var test1 = new String('abc');  // eslint-disable-line
+		test1[5] = 'de';
+		if (Object.getOwnPropertyNames(test1)[0] === '5') {
+			return false;
+		}
```

Comment:
While `Object.getOwnPropertyNames(test1)[0] === '5'` exists in Node v4.0.0 it doesn't effect `Object.assign`.
```js
Object.keys(Object.assign({}, test1))
// => [ '0', '1', '2', '5' ] 
```